src/fuse/arc.rs

```diff
@@ -55,14 +56,29 @@ impl FileSystemAccessor for ArcFuse {
         let read = mode >> 0 & 1;
         let write = mode >> 1 & 1;
         let append = mode >> 2 & 1;
-
+        const REGIONS: &[&str] = &[
+            "jp_ja", "us_en", "us_fr", "us_es", "eu_en", "eu_fr", "eu_es", "eu_de", "eu_nl", "eu_it",
+            "eu_ru", "kr_ko", "zh_cn", "zh_tw",
+        ];
         debug!("Path: {}, read: {}, write: {}, append: {}", path.display(), read, write, append);
+        let mut file_region = Region::UsEnglish;
+        for region in REGIONS.iter() {
+            if path.file_name().unwrap().to_string_lossy().contains(region) {
+                let mut new_path = path.display().to_string();
```

Comment:
Just store the path as a string in a variable and reuse the variable instead, you're converting it twice here.

Reply:
now that u mention it i do see the issue

will fix it when i fix the rest of the stuff